src/main/java/org/jabref/model/entry/AuthorList.java

```diff
@@ -346,51 +326,33 @@ public String getAsNatbibLatexFree() {
      * Oxford comma.</a>
      */
     public String getAsLastNames(boolean oxfordComma) {
-        int abbreviationIndex = oxfordComma ? 0 : 1;
+        return andCoordinatedConjunction(getAuthors(), Author::getLastOnly, oxfordComma);
+    }
 
-        // Check if we've computed this before:
-        if (authorsLastOnly[abbreviationIndex] != null) {
-            return authorsLastOnly[abbreviationIndex];
-        }
+    private static String andCoordinatedConjunction(List<Author> authors, Function<Author, String> style, boolean oxfordComma) {
```

Comment:
One idea, not sure if that is a good one. One could maybe implement this as a stream collector? The usage would then look something like: 
```
getAuthors().stream()
   .map(Author::getLastOnly)
   .collect(andCoordinatedConjunction(oxfordComma));    
```
What do you think? Is this flexible enough?

Reply:
I like the idea of having it as a stream collector. I don't think I were aware of their existence when I wrote https://github.com/JabRef/jabref/pull/6552#issuecomment-641377508.

Your suggestion of extending List would go well with that,
```
AuthorList.of(String).stream()
    .map(Author::getLastOnly)
    .collect(andCoordinatedConjunction(oxfordComma));
```
or
```
AuthorList.of(String).latexFree().stream()
    .map(Author::getLastOnly)
    .collect(andCoordinatedConjunction(oxfordComma));
```